pkg/utils/cmd/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,182 @@
+}
+
+func getConfigmapData(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) ([]byte, error) {
+
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	data, ok := cfgMap.Data["release"]
+	if !ok {
+		return nil, fmt.Errorf("fail to locate \"release\" key in configmap %s/%s", configMapNamespace, configMapName)
+	}
+	return []byte(data), nil
+}
+
+func decodeRelease(data []byte) (*rspb.Release, error) {
+	// base64 decode string
+	b, err := b64.DecodeString(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.Equal(b[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil, err
+		}
+		b2, err := ioutil.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+		b = b2
+	}
+
+	var rls rspb.Release
+	// unmarshal protobuf bytes
+	if err := proto.Unmarshal(b, &rls); err != nil {
+		return nil, err
+	}
+	return &rls, nil
+}
+
+func encodeRelease(rls *rspb.Release) (string, error) {
+	b, err := proto.Marshal(rls)
+	if err != nil {
+		return "", err
+	}
+	var buf bytes.Buffer
+	w, err := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	if err != nil {
+		return "", err
+	}
+	if _, err = w.Write(b); err != nil {
+		return "", err
+	}
+	w.Close()
+
+	return b64.EncodeToString(buf.Bytes()), nil
+}
+
+func updateConfigMap(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace, updatedData string) error {
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	cfgMap.Data["release"] = updatedData
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
+		_, err = k8sClient.CoreV1().ConfigMaps(configMapNamespace).Update(cfgMap)
+		return
+	})
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
+
+	k8sClient, err := BuildClient(*kubeconfig)
+	if err != nil {
+		logrus.Errorf("Failed to create a client: %+v\n", err)
+		os.Exit(1)
+	}
+	if *configMapName == "" {
+		logrus.Errorf("configmap name is missing.")
+		os.Exit(1)
+	}
+	if *configMapNamespace == "" {
+		*configMapNamespace = v1.NamespaceDefault
+	}
+	data, err := getConfigmapData(k8sClient, *configMapName, *configMapNamespace)
+	if err != nil {
+		logrus.Errorf("failed to get Configmap's Data with error: %+v", err)
+		os.Exit(1)
+	}
+	release, err := decodeRelease(data)
+	if err != nil {
+		logrus.Errorf("Fail to decode Release Data of configmap with error: %+v", err)
+		os.Exit(1)
+	}
+	var updatedTemplate []*chart.Template
+	for _, template := range release.Chart.Templates {
+		if template.Name == "templates/crds.yaml" || template.Name == "templates/install-custom-resources.sh.tpl" {
```

Comment:
I am pretty sure we want/need to keep this second manifest.  Would recommend encoding the template names list in a variable and iterating over it.

Reply:
@sdake I would prefer you prove that removing second manifest breaks something. As per iterating comment, it does exactly that, it ranges through the templates.

Reply:
what I had meant is to create an array of template names rather than having a (in the future) bunch of ||s.

somethign like

````
for _, templateName := range templateNames:
    if template.Name == "template/crds.yaml"
        do_work
```

This was sort of in line with Costin's comment about targeted removal of information from the Helm database during an upgrade process.  I believe we had the same thinking (this can be used to migrate to costin's fork of the istio helm repo where the helm charts are broken out.